### PR TITLE
Add missing troubleshooting items

### LINKED ISSUE #141: Missing alarm subcodes in Troubleshooting
The assertion subcodes stop at [55]. The following aren't documented.

```
    SUBCODE_FAIL_CREATE_IO_EXECUTOR = 56,
    SUBCODE_FAIL_TIMER_INIT_USERLAN_MONITOR,
    SUBCODE_FAIL_TIMER_ADD_USERLAN_MONITOR,
    SUBCODE_CONFIGURATION_AGENT_ON_NET_CHECK,

    SUBCODE_CONFIGURATION_FAIL_MP_NICDATA0,
    SUBCODE_CONFIGURATION_FAIL_MP_NICDATA1,
    SUBCODE_FAIL_MP_NICDATA_INIT1,
```

### PR DESCRIPTION
**Pull request by ted-miller**
Fixes #141

### CODE REVIEW

File: doc/troubleshooting.md [hunk 1]
```diff
@@ -582,6 +582,32 @@ The name must not be blank.
 
 After correcting the configuration, the [changes will need to be propagated to the Yaskawa controller](../README.md#updating-the-configuration).
 
+### Alarm: 8011[23 - 54]
+
+*Example:*
+
+```text
+ALARM 8011
+ Failed creating/adding/init *something*
```
Comment (gavanderhoorn) [inline doc/troubleshooting.md:591]: Hmm ..

Could we maybe come up with something a bit nicer? :)

I agree we shouldn't duplicate this entry for all possible subcodes, but perhaps we can come up with a slightly nicer example text?

File: doc/troubleshooting.md [hunk 2]
```diff
@@ -582,6 +582,32 @@ The name must not be blank.
...
+[23] - [54]
+```
+
+*Solution:*
+Ensure that the ROS2 version (Galactic, Humble, etc) is the same for MotoROS2, the micro-ROS Agent and the client PC.
+Additionally, ensure that the ROS2 version is one that is [officially supported](https://github.com/Yaskawa-Global/motoros2#general-requirements) by MotoROS2.
```
Comment (gavanderhoorn) [inline doc/troubleshooting.md:597]:
```suggestion
These alarms are often caused by version incompatibilities between ROS 2 (on the client PC), micro-ROS (as part of MotoROS2) and/or the micro-ROS Agent.

Ensure only compatible versions are used.
As an example: the *Humble* version of MotoROS2 should only be used with ROS 2 *Humble* on the client PC and with the *Humble* version of the micro-ROS Agent.

Please also verify the client PC uses a version of ROS 2 that is [supported by MotoROS2](https://github.com/Yaskawa-Global/motoros2#general-requirements).
```

File: doc/troubleshooting.md [hunk 3]
```diff
@@ -582,6 +582,32 @@ The name must not be blank.
...
+
+If the behavior persists, save a copy of the [debug-listener script](#debug-log-client) output and the `PANELBOX.LOG` from the robot's teach pendant.
```
Comment (gavanderhoorn) [inline doc/troubleshooting.md:599]: Would we want to update all references to the debug logger? This is the first time we link to that section like this.

File: doc/troubleshooting.md [hunk 4]
```diff
@@ -582,6 +582,32 @@ The name must not be blank.
...
+Open a new issue on the [Issue tracker](https://github.com/yaskawa-global/motoros2/issues).
+Describe the problem and include the following items:
+
+- `PANELBOX.LOG`
+- output from debug listener script (complete and unedited)
+- version of MotoROS2
```
Comment (gavanderhoorn) [inline doc/troubleshooting.md:605]: This should already be clear from `PANELBOX.LOG`.

File: doc/troubleshooting.md [hunk 5]
```diff
@@ -582,6 +582,32 @@ The name must not be blank.
...
+- version of micro-ROS Agent
+- version of ROS2 on client PC
```
Comment (gavanderhoorn) [inline doc/troubleshooting.md:607]:
```suggestion
- version of ROS 2 on client PC
```

File: doc/troubleshooting.md [hunk 6]
```diff
@@ -582,6 +582,32 @@ The name must not be blank.
...
+- copy of the MotoROS2 yaml configuration file
```
Comment (gavanderhoorn) [inline doc/troubleshooting.md:608]:
```suggestion
- copy of `motoros2_config.yaml` copied from the robot controller
```

we have some text in the readme detailing how to do this, but it's not in its own section, so we can't directly link to it.

File: doc/troubleshooting.md [hunk 7]
```diff
@@ -609,6 +635,58 @@ Refer to [FAQ: Can names of joints be changed?](faq.md#can-names-of-joints-be-ch
 
 After correcting the configuration, the [changes will need to be propagated to the Yaskawa controller](../README.md#updating-the-configuration).
 
+### Alarm: 8011[56 - 58]
```
Comment (gavanderhoorn) [inline doc/troubleshooting.md:638]: This is basically the same as for subcodes `[23 - 54]`, right?

Would it be acceptable to merge this section with the previous one?

File: doc/troubleshooting.md [hunk 8]
```diff
@@ -609,6 +635,58 @@ Refer to [FAQ: Can names of joints be changed?](faq.md#can-names-of-joints-be-ch
...
+*Example:*
+
+```text
+ALARM 8011
+ Host on NIC check x
+[59]
+```
+
+*Solution:*
+
```
Comment (gavanderhoorn) [inline doc/troubleshooting.md:689]: should this be moved to before the previous section?

Also: is it complete? :)

**Review by gavanderhoorn — COMMENTED** [review]
Round of comments.

File: doc/troubleshooting.md (hunk 1 above)
Comment (ted-miller) [reply doc/troubleshooting.md:591]:
```
ALARM 8011
 MotoROS2: Fatal Error
 Failed adding
 Failed to init
 Failed creating
```

?

File: doc/troubleshooting.md (hunk 4 above)
Comment (ted-miller) [reply doc/troubleshooting.md:605]: Yeah, but it's helpful to see if there are any discrepancies between what someone thinks they're using and what's actually installed.

File: doc/troubleshooting.md (hunk 7 above)
Comment (ted-miller) [reply doc/troubleshooting.md:638]: Unfortunately, that would make [55] non-sequential. 
That would definitely cause issues for someone like me.

File: doc/troubleshooting.md (hunk 8 above)
Comment (ted-miller) [reply doc/troubleshooting.md:689]: I'm not sure what happened here...

**Comment by gavanderhoorn** [comment]
@ted-miller: I've added some commits to polish a few things.

File: doc/troubleshooting.md [hunk 9]
```diff
@@ -609,6 +660,47 @@ Refer to [FAQ: Can names of joints be changed?](faq.md#can-names-of-joints-be-ch
...
+Refer to the relevant Yaskawa Motoman documentation for information on how to change the controller's network configuration.
+
+### Alarm: 8011[60 - 62]
+
+*Example:*
+
+```text
+ALARM 8011
+ Must enable ETHERNET function
+[60 - 61]
```
Comment (gavanderhoorn) [inline doc/troubleshooting.md:697]: is it `[60 - 62]` as in the section header, or `[60 - 61]` as stated here.

Comment (ted-miller) [reply doc/troubleshooting.md]: 62

**Comment by ted-miller** [comment]
I'm find with the edits you made.

**Comment by gavanderhoorn** [comment]
Wrong button :/

feel free to merge @ted-miller.

**Comment by gavanderhoorn** [comment]
Might want to use a squash-merge to prevent all the fixups from getting into the history.

**Pull request merged**